plugins/entando-plugin-jacms/src/main/java/com/agiletec/plugins/jacms/aps/system/services/resource/IResourceManager.java

```diff
@@ -128,6 +128,15 @@
      */
     public void updateResource(ResourceDataBean bean) throws ApsSystemException;
 
+    /**
+     * Clone una risorsa nel db, indipendentemente dal tipo.
+     *
+     * @param resource La risorsa da clonare.
+     * @return la risorsa aggiunta.
+     * @throws ApsSystemException in caso di errore.
+     */
+    public ResourceInterface cloneResource(ResourceInterface resource) throws ApsSystemException;
```

Comment:
You should not change ResourceManager for this feature.
All cloning business logic should be inside ResourcesService and then just call ResourceManager.addResource